cpp/src/arrow/dataset/file_csv.cc

```diff
@@ -52,6 +53,9 @@ using internal::Executor;
 using internal::SerialExecutor;
 
 namespace dataset {
+namespace {
+inline bool parseBool(const std::string& value) { return value == "true" ? true : false; }
```

Comment:
note: please follow the naming convention (UpperCamelCase functions/classes, snake_case variable names)